### PR TITLE
python3.pkgs.thinc: fix build

### PR DESCRIPTION
**Pull request by benxiao**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/thinc/default.nix [hunk 1]
```diff
@@ -71,6 +71,10 @@ buildPythonPackage rec {
     pytestCheckHook
   ];
 
+  postPatch = ''
+    substituteInPlace setup.cfg --replace "blis>=0.7.8,<0.8.0" "blis"
```
Comment (bcdarwin) [inline pkgs/development/python-modules/thinc/default.nix:75]: Note you can often use `pythonRelaxDepsHook` to relax or remove a dependency.

**Review by bcdarwin — APPROVED** [review]
LGTM; builds fine on x86_64-NixOS.

File: pkgs/development/python-modules/thinc/default.nix (hunk 1 above)
Comment (benxiao) [reply pkgs/development/python-modules/thinc/default.nix:75]: I tried that. but it doesn't seem to affect setup.cfg.

Comment (danieldk) [reply pkgs/development/python-modules/thinc/default.nix:75]: Please don't do this. BLIS 0.9.x is broken.

https://github.com/explosion/thinc/issues/771#issuecomment-1255825935
https://github.com/NixOS/nixpkgs/pull/207070

Comment (benxiao) [reply pkgs/development/python-modules/thinc/default.nix:75]: thanks for spotting this. I will close it now.

Comment (benxiao) [reply pkgs/development/python-modules/thinc/default.nix:75]: @danieldk, shall we revert blis to  version 0.7.9, so it is breaking spacy?

Comment (benxiao) [reply pkgs/development/python-modules/thinc/default.nix:75]: https://github.com/NixOS/nixpkgs/pull/223722

**Pull request closed without merging**